### PR TITLE
#676537 setfield value list implementation

### PR DESCRIPTION
**Pull request by Manish-Josh-pepperi**

### CODE REVIEW

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts [hunk 1]
```diff
@@ -1,15 +1,165 @@
...
+      },
+    };
   }
 
+  private async handleEditAction(selectedData: PepSelectionData) {
+    const selectedRowID = selectedData.rows[0];
+    const selectedItem = this._currentConfiguration.find(item => item.Resource === selectedRowID);
+    this.hostObject.Configuration = selectedItem;
+    // this.dialogService.openDialog(
+    //   this.translate.instant("SEARCH_DATA.TITLE"),
```
Comment (izhar-ahmad-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:114]: Remove Commented code, multiple places

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts [hunk 2]
```diff
@@ -1,15 +1,165 @@
...
+    //   SearchDataLogicBlockComponent,
+    //   [],
+    //   this.hostObject,
+    //   this.addResourceCB
+    // );
+  }
+
+  private async handleDeleteAction(selectedData: PepSelectionData) {
+    const selectedRowID = selectedData.rows[0];
+    const indexToRemove = this._currentConfiguration.findIndex(item => item.Resource === selectedRowID);
```
Comment (izhar-ahmad-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:124]: Is Resource unique for each item?

File: shared/build/shared/index.js [hunk 3]
```diff
@@ -20,4 +20,5 @@ __exportStar(require("./get-values.model"), exports);
 __exportStar(require("./open-external.model"), exports);
 __exportStar(require("./search-data.model"), exports);
 __exportStar(require("./helper"), exports);
+__exportStar(require("./set-field-value.model"), exports);
```
Comment (izhar-ahmad-pepperi) [inline shared/build/shared/index.js:23]: Why are there changes from build?

**Review by izhar-ahmad-pepperi — CHANGES_REQUESTED** [review]

File: shared/build/shared/index.d.ts
Comment (roiaharon-pepperi) [inline shared/build/shared/index.d.ts:1]: build folder shouldn't be commited.. please add this folder to the git.ignore file

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts (hunk 1 above)
Comment (Manish-Josh-pepperi) [reply client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:114]: These are going to be used in the next PR. I have enabled these lines of code.
in another PR.

File: shared/set-field-value.model.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
+export interface SetFieldValueConfiguration {
```
Comment (roiaharon-pepperi) [inline shared/set-field-value.model.ts:2]: isn't this interface is for the searchData? where is the path parameters? and other configuration?

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts (hunk 2 above)
Comment (roiaharon-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:124]: I'm not sure what exactly are you doing here.. didn't you generated a uuid for each line? why not using it?
resource is not unique here (maybe {{resource}}_{{fieldName}}

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts (hunk 2 above)
Comment (Manish-Josh-pepperi) [reply client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts]: Yes, that has been taken care of in the next task PR.

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts [hunk 5]
```diff
@@ -1,15 +1,165 @@
...
+  // ngOnInit(): void {
+  // }
+  
+  // Getter/Setter
+  get currentConfiguration(): SetFieldValueConfiguration {
+    return this._currentConfiguration as SetFieldValueConfiguration;
+  }
+
+  async run() {
+    const filteredConfig = this._currentConfiguration.filter(config => config.Resource !== "");
```
[the remark this answers is not in the document]
Comment (Manish-Josh-pepperi) [reply client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts]: this has been taken care of in the next PR.

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.scss [hunk 6]
```diff
@@ -0,0 +1,5 @@
+.configuration-content {
+  min-height: 32rem;
```
Comment (roiaharon-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.component.scss:2]: I have a deja voo, why is this style here? isn't this a dialog?

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts [hunk 7]
```diff
@@ -45,17 +45,31 @@ export class SetFieldValueLogicBlockComponent extends BaseLogicBlockDirective {
...
+    debugger;
+    const filteredConfig = this._currentConfiguration.filter(config => config.Source !== "");
+    filteredConfig.forEach(config => this.resourceSet.add(config.Source));
     this.listDataSource = this.getDataSource(filteredConfig); 
     this.isLoaded = true;
   }
 
   reload() {
-    const filteredConfig = this._currentConfiguration.filter(config => config.Resource !== "");
+    debugger;
```
Comment (roiaharon-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:56]: remove debugger

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts [hunk 8]
```diff
@@ -45,17 +45,31 @@ export class SetFieldValueLogicBlockComponent extends BaseLogicBlockDirective {
   }
 
   async run() {
-    const filteredConfig = this._currentConfiguration.filter(config => config.Resource !== "");
-    filteredConfig.forEach(config => this.resourceSet.add(config.Resource));
+    debugger;
+    const filteredConfig = this._currentConfiguration.filter(config => config.Source !== "");
+    filteredConfig.forEach(config => this.resourceSet.add(config.Source));
```
Comment (roiaharon-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:50]: you are already looping through all the items with the filter, why not do this in the same callback?

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts [hunk 9]
```diff
@@ -121,11 +128,11 @@ export class SetFieldValueLogicBlockComponent extends BaseLogicBlockDirective {
...
     const selectedRowID = selectedData.rows[0];
-    const indexToRemove = this._currentConfiguration.findIndex(item => item.Resource === selectedRowID);
+    const indexToRemove = this._currentConfiguration.findIndex(item => item.Key === selectedRowID);
     if (indexToRemove !== -1) {
       this._currentConfiguration.splice(indexToRemove, 1);
-      // this.resourceSet.delete(selectedRowID);
-      // this.reload();
+      this.resourceSet.delete(selectedRowID);
+      this.reload();
       this.hostObject.Configuration = this._currentConfiguration;
```
Comment (roiaharon-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:136]: why are you setting the host object? this should be read only. if you want to update the configurations there are output callbacks to achieve that

File: .gitignore [hunk 10]
```diff
@@ -4,4 +4,5 @@ cpi-side/build
 server-side/build
 **/node_modules/**
 .DS_Store
-build
\ No newline at end of file
+shared/build
```
Comment (roiaharon-pepperi) [inline .gitignore:7]: this still not working.. there are still build files in the commit..
this is the git.ignore from other project, maybe try to use that
publish
var_sk
\*\*/node_modules/\*\*
\*\*/build/\*\*

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts (hunk 8 above)
Comment (Manish-Josh-pepperi) [reply client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:50]: all this taken care in next PR #16

**Review by roiaharon-pepperi — APPROVED** [review]

**Pull request merged**